haystack/document_store/milvus.py

```diff
@@ -0,0 +1,420 @@
+                vector_ids_for_query.append(str(vector_id))
+                scores_for_vector_ids[str(vector_id)] = distance
+
+        documents = self.get_documents_by_vector_ids(vector_ids_for_query, index=index)
+        for doc in documents:
+            doc.score = scores_for_vector_ids[doc.meta["vector_id"]]
+            doc.probability = float(expit(np.asarray(doc.score / 100)))
+            if return_embedding is True:
+                doc.embedding = self._get_embedding_by_id(index=index, id=doc.meta.get("vector_id"))
+
+        return documents
+
+    def delete_all_documents(self, index: Optional[str] = None, filters: Optional[Dict[str, List[str]]] = None):
+        index = index or self.index
+        super().delete_all_documents(index=index, filters=filters)
+        status, ok = self.milvus_server.has_collection(collection_name=index)
+        if status.code != Status.SUCCESS:
+            raise RuntimeError(f'Milvus has collection check failed: {status}')
+        if ok:
+            status = self.milvus_server.drop_collection(collection_name=index)
+            if status.code != Status.SUCCESS:
+                raise RuntimeError(f'Milvus drop collection failed: {status}')
+
+            self.milvus_server.flush([index])
+            self.milvus_server.compact(collection_name=index)
+
+    def get_all_documents_generator(
+        self,
+        index: Optional[str] = None,
+        filters: Optional[Dict[str, List[str]]] = None,
+        return_embedding: Optional[bool] = None,
+        batch_size: int = 10_000,
+    ) -> Generator[Document, None, None]:
+        """
+        Get all documents from the document store. Under-the-hood, documents are fetched in batches from the
+        document store and yielded as individual documents. This method can be used to iteratively process
+        a large number of documents without having to load all documents in memory.
+
+        :param index: Name of the index to get the documents from. If None, the
+                      DocumentStore's default index (self.index) will be used.
+        :param filters: Optional filters to narrow down the documents to return.
+                        Example: {"name": ["some", "more"], "category": ["only_one"]}
+        :param return_embedding: Whether to return the document embeddings.
+        :param batch_size: When working with large number of documents, batching can help reduce memory footprint.
+        """
+        index = index or self.index
+        documents = super().get_all_documents_generator(
+            index=index, filters=filters, batch_size=batch_size
+        )
+        if return_embedding is None:
+            return_embedding = self.return_embedding
+
+        for doc in documents:
+            if return_embedding:
+                if doc.meta and doc.meta.get("vector_id") is not None:
+                    doc.embedding = self._get_embedding_by_id(index=index, id=str(doc.meta.get("vector_id")))
```

Comment:
We could consider batching this to avoid a large number of network calls to Milvus.